docs/rules/no-unused-modules.md

```diff
@@ -0,0 +1,80 @@
+# import/no-unused-modules
+
+Reports:
+  - modules without any or
+  - individual exports not being used within other modules
+
+## Rule Details
+
+
+### Options
+
+This rule takes the following option:
+
+- `src`: an array with files/paths to be analyzed. It only for applies for unused exports  
+- `ignore`: an array with files/paths to be ignored. It only for applies for unused exports  
+- `missingExports`: if `true`, files without any exports are reported  
+- `unusedExports`: if `true`, exports without any usage are reported
```

Comment:
Thanks for the input. I will update the documentation.

Reply:
I'm confused, why would you ever want to allow an export that's only used in the same file and never imported?

Reply:
I was misunderstanding the purpose of the `ignore` option.  I had expected that `ignore: ['test/**']` would tell this rule to ignore imports performed by tests - thus report anything which is not imported from production code.  A common pattern in my applications is for internal sources to contain `export function defaultOpts() { return {...}; }`, and for that function to be used in exactly two places: the constructor function in the same source and the unit test.

To be honest I don't understand the purpose of the `ignore` option as it's currently implemented.  Why not just use .eslintrc `overrides` to turn this rule off for certain files?  Please correct me if I'm wrong but the `src` option doesn't tell this rule which files to lint, it tells it which files scan for import's, then each file being linted is checked to make sure something from `src` performed the import?  If so I think that `ignore` should counter-act `src`, meaning restrict the list of files scanned for import's - not restrict the list of files to lint.

Reply:
The option is for package entry points, which are often untested (because they just re-export something from the build dir).

I expect the spec option to cover files whose *exports* should be checked. Anything that imports it anywhere should count it as used.

Reply:
As @ljharb said, the ```ignore``` option is mainly for package entry points. Every file specified in this option will not be linted by this rule. 
However, these files are also analyzed to handle their imports. This means, if an ignored file imports from anywhere else, let's say from ```file-a.js``` the corresponding exports from file  ```file-a.js```  will still be marked as used.

Reply:
I misunderstood the purpose of the `ignore` option, IMO it would be clearer if the docs mentioned package entry points.  Renaming it to `ignoreExports` or `entryPoints` could make it's purpose even clearer.

Would you accept a follow-up patch to implement an `ignoreImports` option?  This would allow me to identify when an API is no longer used even though it's still covered by unit tests.  I do test driven development so every export will be imported (directly or indirectly) by unit tests.

Thank you for your responses!

Reply:
I agree; both options are needed to be maximally useful.

Reply:
Nothing was ever changed in response to this comment thread.  I'm fine with pushing a follow-up PR to add support for ignoring importers (like tests for example), but I think naming the option `ignore` is confusing.

If you don't want to rename the option maybe some rewording of the docs?  Especially the docs for `ignore`, anyone trying to ignore tests will see the current docs and assume that it allows them to ignore importers (tests), rather than ignoring module entry-points.
```
- `src`: an array with files/paths with exports that must be used.  Defaults to `process.cwd()`, if not provided
- `ignore`: an array with files/paths where unused exports are ignored.
```

Reply:
@coreyfarrell: sorry, that wasn't intentionally. I will rename the ```ignore```-option to ```ignoreImports``` and do some rewording in the docs.